src/routes/business/business-email-check-routes.js

```diff
@@ -1,26 +1,23 @@
+import { fetchBusinessEmailChangeService } from '../../services/business/fetch-business-email-change-service.js'
+import { updateBusinessEmailChangeService } from '../../services/business/update-business-email-change-service.js'
+import { businessEmailChangePresenter } from '../../presenters/business/business-email-change-presenter.js'
 const getBusinessEmailCheck = {
   method: 'GET',
   path: '/business-email-check',
-  handler: (request, h) => {
-    const businessEmail = request.state.businessEmail || ''
+  handler: async (request, h) => {
+    const businessEmailChange = await fetchBusinessEmailChangeService(request.yar)
+    const pageData = businessEmailChangePresenter(businessEmailChange, request.yar)
 
-    return h.view('business/business-email-check', {
-      businessEmail
-    })
+    return h.view('business/business-email-check.njk', pageData)
   }
 }
 
 const postBusinessEmailCheck = {
   method: 'POST',
   path: '/business-email-check',
-  handler: (request, h) => {
-    const businessEmail = request.state.businessEmail
-
+  handler: async (request, h) => {
+    await updateBusinessEmailChangeService(request.yar)
```

Comment:
This is where you want to set the yar notification for the business-details page to confirm the update made.

```suggestion
    await updateBusinessEmailChangeService(request.yar)
    
    flashNotification(request.yar, 'Success', 'You have updated your business email')
```